File: app/src/main/java/com/tutv/android/datasource/retrofit/endpoint/NetworksAPI.java
package com.tutv.android.datasource.retrofit.endpoint;

import com.tutv.android.domain.Network;

import java.util.List;

import io.reactivex.Single;
import retrofit2.http.GET;

public interface NetworksAPI {
    @GET("networks")
    Single<List<Network>> getAll();
}

File: app/src/main/java/com/tutv/android/datasource/retrofit/endpoint/NetworksAPI.kt
package com.tutv.android.datasource.retrofit.endpoint

import com.tutv.android.datasource.retrofit.annotation.AuthenticatedRequest
import com.tutv.android.domain.Network
import io.reactivex.Single
import org.mockito.ArgumentMatchers
import org.mockito.InOrder
import org.mockito.Mockito
import retrofit2.http.GET

interface NetworksAPI {
    @AuthenticatedRequest
    @GET("networks")
    open fun getAll(): Single<MutableList<Network?>?>?
}